2009-10-10  Jakub Jelinek  <jakub@redhat.com>

	* config/rs6000/sysv4.h (TARGET_DEFAULT): Override to default to
	power4 for RHEL6.
	* config/rs6000/default64.h (TARGET_DEFAULT): Likewise.

--- gcc/config/rs6000/sysv4.h	2009-10-07 16:45:42.000000000 +0200
+++ gcc/config/rs6000/sysv4.h	2009-10-10 15:48:38.197497571 +0200
@@ -239,9 +239,10 @@ do {									\
 } while (0)
 #endif
 
-/* Override rs6000.h definition.  */
+/* Override rs6000.h definition.  RHEL6 defaults to power4 CPU, power6 tuning.  */
 #undef	TARGET_DEFAULT
-#define	TARGET_DEFAULT (MASK_POWERPC | MASK_NEW_MNEMONICS)
+#define	TARGET_DEFAULT (MASK_POWERPC | MASK_NEW_MNEMONICS | MASK_POWERPC64 \
+			| MASK_PPC_GPOPT | MASK_PPC_GFXOPT | MASK_MFCRF)
 
 /* Override rs6000.h definition.  */
 #undef	PROCESSOR_DEFAULT
--- gcc/config/rs6000/default64.h	2009-01-14 12:06:23.000000000 +0100
+++ gcc/config/rs6000/default64.h	2009-10-10 15:49:44.533872481 +0200
@@ -18,7 +18,9 @@ You should have received a copy of the G
 along with GCC; see the file COPYING3.  If not see
 <http://www.gnu.org/licenses/>.  */
 
+/* RHEL6 defaults to power4 CPU, power6 tuning.  */
 #undef TARGET_DEFAULT
 #define TARGET_DEFAULT \
   (MASK_POWERPC | MASK_PPC_GFXOPT | \
-   MASK_POWERPC64 | MASK_64BIT | MASK_NEW_MNEMONICS)
+   MASK_POWERPC64 | MASK_64BIT | MASK_NEW_MNEMONICS \
+   MASK_PPC_GPOPT | MASK_MFCRF)
